barretenberg/cpp/pil/avm/avm_main.pil

```diff
@@ -508,6 +518,11 @@ namespace avm_main(256);
     ALU_R_TAG_SEL * (alu_in_tag - r_in_tag) = 0;
     ALU_W_TAG_SEL * (alu_in_tag - w_in_tag) = 0;
 
+    //===== GASLEFT OPCODES =====================================================
+    #[L2GASLEFT]
+    sel_op_l2gasleft * (ia - l2_gas_remaining') = 0;
```

Comment:
so its the gas left after itself has executed?

Reply:
Yes, this is according the yellow paper instruction spec.

Reply:
We checked the simulator code and it matches.